flask-homework/app3.py

```diff
@@ -0,0 +1,198 @@
+from flask import Flask, request, render_template, redirect, abort, session, url_for
+import random
+import os
+
+app = Flask(__name__, template_folder='template_html')
+app.secret_key = os.urandom(24)  # Set a secret key for the session.
+
+
+@app.route('/users')
+def get_users():
+    if 'username' in session:
+        greeting = f'Hello, {session["username"]}'
```

Comment:
nit: better to pass username in context and create the particular rendering in a template using `if` construction.